### PR TITLE
feat: Update "ViewChanges" Commands to use Source Tracking Library

### PR DESCRIPTION
**Pull request by klewis-sfdc**
### What does this PR do?
* Updates the following commands to use the Source Tracking library instead of using the sfdx cli:
  * SFDX: View All Changes (Local and in Default Org)
  * SFDX: View Local Changes
  * SFDX: View Changes in Default Org

### What issues does this PR fix or reference?
@W-10931530@

### Functionality Before
* View Changes commands used cli

### Functionality After
* View Changes commands use source tracking library

### CODE REVIEW

**Review by randi274 — COMMENTED** [review]
Review notes so far!

File: packages/salesforcedx-utils-vscode/src/services/sourceTrackingService.ts [hunk 1]
```diff
@@ -45,4 +49,157 @@ export class SourceTrackingService {
...
+    StatusOutputRow['state'],
+    StatusResult['actualState']
+  >([
+    ['delete', 'Deleted'],
+    ['add', 'Add'],
+    ['modify', 'Changed'],
+    ['nondelete', 'Changed']
+  ]);
+
+  // sort order is state, type, fullname
```
[the remark this answers is not in the document]
Comment (klewis-sfdc) [reply packages/salesforcedx-utils-vscode/src/services/sourceTrackingService.ts:159]: I copied this exactly from cli code when I brought it in so 🤷🏽‍♂️

File: packages/salesforcedx-utils-vscode/src/services/sourceTrackingService.ts [hunk 2]
```diff
@@ -45,4 +49,157 @@ export class SourceTrackingService {
...
+  ]);
+
+  private static stateMap = new Map<
+    StatusOutputRow['state'],
+    StatusResult['actualState']
+  >([
+    ['delete', 'Deleted'],
+    ['add', 'Add'],
+    ['modify', 'Changed'],
+    ['nondelete', 'Changed']
```
[the remark this answers is not in the document]
Comment (klewis-sfdc) [reply packages/salesforcedx-utils-vscode/src/services/sourceTrackingService.ts:156]: Same as above - this is copied code from cli plugins

File: packages/salesforcedx-utils-vscode/src/services/sourceTrackingService.ts [hunk 3]
```diff
@@ -45,4 +49,157 @@ export class SourceTrackingService {
...
+  const workspaceContext = WorkspaceContextUtil.getInstance();
+  const connection = await workspaceContext.getConnection();
+  const sourceTracking = await SourceTrackingService.createSourceTracking(
+    rootWorkspacePath,
+    connection
+  );
+  return sourceTracking;
+}
+
+type StatusActualState = 'Deleted' | 'Add' | 'Changed' | 'Unchanged';
```
[the remark this answers is not in the document]
Comment (klewis-sfdc) [reply packages/salesforcedx-utils-vscode/src/services/sourceTrackingService.ts:80]: We could convert this to an enum but this is also copied code so I'd probably just leave it - this stuff should all be giving us a similarly formatted output table to what the output gives when running `status` from cli.

File: packages/salesforcedx-utils-vscode/test/jest/services/source-tracking/sourceTrackingService.test.ts [hunk 4]
```diff
@@ -52,4 +54,45 @@ describe('Source Tracking Service', () => {
...
+      sourceTrackingMock = jest
+        .spyOn(SourceTracking, 'create')
+        .mockResolvedValue({
+          getStatus: jest
+            .fn()
+            .mockResolvedValue(testData.statusOutputRows as any)
+        } as any);
+    });
+
+    it('Should return a properly formatted string when local and remote changes exist.', async () => {
```
[the remark this answers is not in the document]
Comment (randi274) [reply packages/salesforcedx-utils-vscode/test/jest/services/source-tracking/sourceTrackingService.test.ts:86]: I also don't want this to hold up the work, so I think we could defer here if they also don't have test coverage here.

File: packages/salesforcedx-vscode-core/src/commands/viewChanges.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/src/types';
+import { channelService, OUTPUT_CHANNEL } from '../channels';
+import { nls } from '../messages';
+import {
+  EmptyParametersGatherer,
+  SfdxCommandlet,
+  SfdxWorkspaceChecker
+} from './util';
+
+export class SourceTrackingGetStatusExecutor extends LibraryCommandletExecutor<
```
Comment (randi274) [inline packages/salesforcedx-vscode-core/src/commands/viewChanges.ts:20]: This one does seem net-new for us, so we probably would want to create tests here.

**Review by randi274 — APPROVED** [review]
Completed QA, and discussed with Ken that the commands for View All Changes and View Remote Changes are a little different (missing the "IGNORED" column"). Can confirm we're no longer using the CLI for any of these commands. I defer to him on if that needs to be updated here, or in SDR, or neither. Otherwise QA is looking good - if we get a little more test coverage and finish the label updates I'm good 👍

File: packages/salesforcedx-utils-vscode/src/services/sourceTrackingService.ts [hunk 6]
```diff
@@ -45,4 +50,156 @@ export class SourceTrackingService {
...
+      (this.statusResults as unknown) as Row[],
+      this.columns
+    );
+
+    return table;
+  }
+
+  private convertToTableRow(result: StatusResult): FormattedStatusResult {
+    return Object.assign(result, {
+      ignored: result.ignored ? 'true' : 'false',
```
Comment (klewis-sfdc) [inline packages/salesforcedx-utils-vscode/src/services/sourceTrackingService.ts:201]: Updated this based on Randi's findings from QA.  At this point in time, now matches up with the cli output from force:source:status:

**Review by randi274 — APPROVED** [review]

**Pull request merged**